capytaine/io/xarray.py

```diff
@@ -238,15 +239,44 @@ def assemble_dataset(results: Sequence[LinearPotentialFlowResult],
     """
     dataset = xr.Dataset()
 
+    if hasattr(results, '__iter__'):
+        try:
+            if 'capytaine' in results[0].__module__:
+                bemio_import = False
+            else:
+                raise TypeError('results must be either of type LinearPotentialFlowResult or a bemio.io object')
+        except:
+            raise TypeError('results must be either of type LinearPotentialFlowResult or a bemio.io object')
+
+    else:
+        try:
+            if 'bemio.io' in results.__module__:
+                bemio_import = True
+            else:
+                raise TypeError('results must be either of type LinearPotentialFlowResult or a bemio.io object')
+        except:
+            raise TypeError('results must be either of type LinearPotentialFlowResult or a bemio.io object')
+    
+    if bemio_import:
+        records = dataframe_from_bemio(results)
+        all_dofs_in_order = {'Surge': [1., 0., 0., 0., 0., 0.],
```

Comment:
I don't remember why I used a dict with `None`s as values for `all_dofs_in_order`. A list might have been simpler.
Anyway, do you need the `[1, 0, ...]` values here?

Reply:
They are not needed. I initially put these in as placeholders instead of `None`s, as I wasn't sure at the time whether the dict values were used anywhere, and then neglected to change them. This has been corrected in commit 0f16541.